### DISCUSSION TITLE
Org-table alignment (divider rows)

### DISCUSSION THREAD

**Discussion by seb-hyland** [minad/org-modern#216] [Q&A] [answered]
Org-modern formatted tables typically align very well, but divider rows do not align with standard rows.

For instance, divider rows in org typically look like:
```
|-----------+-----------------------------------------------|
```
where the + aligns with the | divider in rows above/below:
```
| Keybind   | Action                                          |
|-----------+-------------------------------------------------|
| SHIFT-h/l | Jump forwards/backwards by one *word*           |
```
However, in org-modern the divider row is usually too short:
![20240626_14h29m07s_grim](https://github.com/minad/org-modern/assets/158851682/b9fa91a2-b3b8-4c5f-bfdd-2cdeb599d7fd)

I am using a monospaced font for org-table, specifically the [nerd-font patched version of SF Mono](https://github.com/shaunsingh/SFMono-Nerd-Font-Ligaturized). I have also tried Iosevka Aile and JetBrainsMono Nerd Font, and have experienced the same issue.

I'm not sure if there is some configuration that I need to fix on my end; I have tried different values for org-modern-table-horizontal, but the spacing issue remains.

**Comment by minad** [comment]
Hi, is it possible to reproduce this with emacs -Q? Do you have a minimal configuration?

**Comment by seb-hyland** [comment]
I've figured it out; variable-pitch-mode seems to break the table formatting even if org-table is set to a monospaced font. I tried using mixed-pitch mode as well to no success. Do you have any idea how I might be able to fix this? I heavily prefer variable-pitch-mode for document editing.

**Reply by seb-hyland** [reply]
I messed with the face height of org-table, and got it to line up right (see below). However, there seems to be a large gap that I can't figure out how to get rid of (I tried different values of line-spacing).

I don't understand why the table is influenced by variable-pitch-mode; is it dependent on faces other than org-table? Is there any way to make it calculated based solely off of the org-table sizes?

![20240629_01h31m47s_grim](https://github.com/minad/org-modern/assets/158851682/9dca94ca-c85d-48f1-ba0d-5549d72c2015)

**Reply by seb-hyland** [reply]
After doing a little more digging, I swapped out `(default-font-width)` for `(window-font-width (selected-window) 'org-table)` in org-modern.el. This helps make it so I don't have to manually adjust org-table's height to try and match it to the variable pitch font, but I still have the excess spacing issue and the very end of the line isn't properly hidden (see screenshot). Please let me know if you can think of any solutions.

![20240629_02h30m57s_grim](https://github.com/minad/org-modern/assets/158851682/294d067e-b0fd-4f96-9bf4-0adbca9ddf6c)

**Reply by minad** [reply, marked as answer]
Should be fixed by https://github.com/minad/org-modern/commit/1074133d9dd1ce4eb6c9ab1d1dbc7b78b3d8733b
